tests/resources/functional/snowflake/core_engine/lca/lca_nested.sql

```diff
@@ -0,0 +1,13 @@
+-- snowflake sql:
+SELECT
+    b * c as new_b,
+    a - new_b as ab_diff
```

Comment:
For fun, how about we swap those operators:
```suggestion
    b - c as new_b,
    a * new_b as ab_diff
```

Reply:
the behavior might indeed surprise... we'd probably want to raise a warning if dealiasing involves more than 2 expressions

Reply:
See last comment #1276